akka-remote/src/main/scala/akka/remote/security/setup/CryptoServiceProviderSetup.scala

```diff
@@ -0,0 +1,80 @@
+package akka.remote.security.setup
+
+import java.security.{ AccessController, PrivilegedAction, Provider }
+import java.util.Collections.{ emptyList, emptyMap }
+import javax.net.ssl._
+
+import akka.actor.setup.Setup
+import akka.remote.security.provider._
+
+import scala.reflect.ClassTag
+
+abstract class CryptoServiceProviderSetup extends Setup {
+  def provider: Provider
+}
+
+case class KeyManagerFactorySetup(provider: Provider, keyManagerFactoryParameters: Option[ManagerFactoryParameters])
+  extends CryptoServiceProviderSetup
+
+object KeyManagerFactorySetup {
+  /**
+   * Java API
+   */
+  def create(provider: Provider): KeyManagerFactorySetup = KeyManagerFactorySetup(provider, None)
+
+  /**
+   * Java API
+   */
+  def create(keyManagerFactoryProvider: Provider, keyManagerFactoryParameters: ManagerFactoryParameters): KeyManagerFactorySetup =
+    KeyManagerFactorySetup(keyManagerFactoryProvider, Some(keyManagerFactoryParameters))
+
+  // TODO Java version
```

Comment:
Still missing Java version.

Also, I can't seem to find any test coverage for this one.